annotationsx/middleware.py

```diff
@@ -134,4 +134,23 @@ def process_response(self, request, response):
                                 path=settings.SESSION_COOKIE_PATH,
                                 secure=settings.SESSION_COOKIE_SECURE or None,
                                 httponly=settings.SESSION_COOKIE_HTTPONLY or None)
-        return response
\ No newline at end of file
+        return response
+
+
+class CookielessSessionMiddleware(object):
+    def __init__(self):
+        engine = import_module(settings.SESSION_ENGINE)
+        self.SessionStore = engine.SessionStore
+        print "STarting engine"
+
+    def process_request(self, request):
+        session_key = request.COOKIES.get(settings.SESSION_COOKIE_NAME, request.GET.get('utm_source'))
```

Comment:
Oh I see. gotcha. The only reason why we used utm_source was for some very simplistic obfuscation. Passing along sessionIds via params seems like a security nightmare so we were trying to misdirect a bit or at least not be so obvious about it. It's not entirely necessary and wouldn't be too terrible to do your way if you think that's better.